### PR TITLE
Add a default password authentication setting

### PR DESCRIPTION
**Pull request by nod0n**
#### Pull Request (PR) description

This PR enables you to pick your default for password authentication. This is done by passing a value to `$sftp_jail::password_authentication`.

### CODE REVIEW

File: spec/defines/sftp_jail_jail_spec.rb [hunk 1]
```diff
@@ -56,6 +56,28 @@
...
       end
+
+      let :params do
+        {
+          user: 'alice',
+          group: 'alice',
+          password_authentication: 'yes'
+        }
+      end
+      let :title do
```
Comment (bastelfreak) [inline spec/defines/sftp_jail_jail_spec.rb:67]: maybe this needs to be wrapped in a new context block so title can be overwritten?

Comment (nod0n) [reply spec/defines/sftp_jail_jail_spec.rb:75]: I tried here: https://github.com/voxpupuli/puppet-sftp_jail/pull/74/commits/6c5633152a8317735a582928c7da5083411b8542 didn't help. :(

File: spec/defines/sftp_jail_jail_spec.rb [hunk 2]
```diff
@@ -57,5 +57,37 @@
...
+        'test2'
+      end
+
+      it 'allows PasswordAuthentication for the user alice' do
+        is_expected.to contain_ssh__server__match_block('alice').with(
+          'type' => 'Group',
+          'options' => {
+            'ChrootDirectory'        => '/chroot/test',
```
Comment (bastelfreak) [inline spec/defines/sftp_jail_jail_spec.rb:83]: this should be:
```suggestion
            'ChrootDirectory'        => '/chroot/test2',
```

?

**Comment by nod0n** [comment]
This is now backwards compatible.

**Pull request merged**